code/modules/spells/spell_types/pointed/swap.dm

```diff
@@ -0,0 +1,46 @@
+
+/datum/action/cooldown/spell/pointed/swap
+	name = "Swap"
+	desc = "This spell instills a deep terror in your target, temporarily chilling and blinding it."
+	button_icon_state = "swap"
+	ranged_mousepointer = 'icons/effects/mouse_pointers/swap_target.dmi'
+
+	school = SCHOOL_EVOCATION
+	cooldown_time = 30 SECONDS
+	cooldown_reduction_per_rank = 6 SECONDS
```

Comment:
5 levels of this gives a cooldown time of 0 seconds, have you tested to make sure this won't happen in game?

Reply:
The rank starts at 30, so level five is minus 24 = 6